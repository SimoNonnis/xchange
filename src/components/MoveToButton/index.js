import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  code: PropTypes.string.isRequired,
  onClick: PropTypes.func
};

const defaultProps = {
  onClick: () => undefined
};

const MoveToButton = ({ code, onClick }) => (
  <button className="move-to-button" onClick={onClick}>
    {code}
  </button>
);

MoveToButton.propTypes = propTypes;
MoveToButton.defaultProps = defaultProps;

export default MoveToButton;
