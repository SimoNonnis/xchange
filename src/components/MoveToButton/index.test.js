import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import MoveToButton from './index';

describe('Test <MoveToButton />', () => {
  it('should render with required props', () => {
    const wrapper = shallow(<MoveToButton code="RUB" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render with all props', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<MoveToButton code="RUB" onClick={onClick} />);

    wrapper.simulate('click');
    expect(onClick).toHaveBeenCalled();
  });
});
