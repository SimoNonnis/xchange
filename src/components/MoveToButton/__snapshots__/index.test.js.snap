// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <MoveToButton /> should render with required props 1`] = `
<button
  className="move-to-button"
  onClick={[Function]}
>
  RUB
</button>
`;
